cli/azd/pkg/infra/provisioning/terraform/terraform_provider.go

```diff
@@ -65,37 +56,33 @@ func (t *TerraformProvider) RequiredExternalTools() []tools.ExternalTool {
 
 // NewTerraformProvider creates a new instance of a Terraform Infra provider
 func NewTerraformProvider(
-	ctx context.Context,
+	cli terraform.TerraformCli,
 	env *environment.Environment,
-	projectPath string,
-	infraOptions Options,
 	console input.Console,
-	commandRunner exec.CommandRunner,
 	curPrincipal CurrentPrincipalIdProvider,
-	prompters Prompters,
-) *TerraformProvider {
-	terraformCli := terraform.NewTerraformCli(commandRunner)
-
-	// Default to a module named "main" if not specified.
-	if strings.TrimSpace(infraOptions.Module) == "" {
```

Comment:
I think you need to move L80-L83 into Initialize as you had done in the bicep provider?

Reply:
Ahh - you did this as I was reviewing to fix an end to end test.  Good on us for having a test to catch this!